Configure fails when MOAB is enabled but MOAB_LIBRARY_DIRS are not specified

I have a simple
```cmake
-D TPL_ENABLE_MOAB=ON
```
in my configuration script. It fails with
```
touch: cannot touch '/moab-revision': Permission denied
CMake Error at DataTransferKit/CMakeLists.txt:99 (FILE):
  file failed to open for reading (No such file or directory):

    /moab-revision


MOAB Revision = ''
```
The issue seems to be obvious: I don't provide `MOAB_LIBRARY_DIRS` as it finds it fine without.

The workaround so far is
```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 04f6e22..4479aa0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -93,13 +93,13 @@ IF(GIT_FOUND)
     MESSAGE("DTK Revision = '${DTK_REVISION}'")
 ENDIF()
 
-IF(Trilinos_ENABLE_DataTransferKitMoabAdapters)
-    SET(MOAB_STAMP_FILE ${MOAB_LIBRARY_DIRS}/moab-revision)
-    EXECUTE_PROCESS(COMMAND touch ${MOAB_STAMP_FILE})
-    FILE(READ ${MOAB_STAMP_FILE} MOAB_REVISION)
-    STRING(REGEX REPLACE "\n" "" MOAB_REVISION "${MOAB_REVISION}")
-    MESSAGE("MOAB Revision = '${MOAB_REVISION}'")
-ENDIF()
+# IF(Trilinos_ENABLE_DataTransferKitMoabAdapters)
+    # SET(MOAB_STAMP_FILE ${MOAB_LIBRARY_DIRS}/moab-revision)
+    # EXECUTE_PROCESS(COMMAND touch ${MOAB_STAMP_FILE})
+    # FILE(READ ${MOAB_STAMP_FILE} MOAB_REVISION)
+    # STRING(REGEX REPLACE "\n" "" MOAB_REVISION "${MOAB_REVISION}")
+    # MESSAGE("MOAB Revision = '${MOAB_REVISION}'")
+# ENDIF()
 
 ##---------------------------------------------------------------------------##
 ## Check C++ code style using clang-format version 3.9
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -93,14 +93,6 @@ IF(GIT_FOUND)
     MESSAGE("DTK Revision = '${DTK_REVISION}'")
 ENDIF()
 
-IF(Trilinos_ENABLE_DataTransferKitMoabAdapters)
-    SET(MOAB_STAMP_FILE ${MOAB_LIBRARY_DIRS}/moab-revision)
-    EXECUTE_PROCESS(COMMAND touch ${MOAB_STAMP_FILE})
-    FILE(READ ${MOAB_STAMP_FILE} MOAB_REVISION)
-    STRING(REGEX REPLACE "\n" "" MOAB_REVISION "${MOAB_REVISION}")
-    MESSAGE("MOAB Revision = '${MOAB_REVISION}'")
-ENDIF()
-
 ##---------------------------------------------------------------------------##
 ## Check C++ code style using clang-format version 3.9
 ##---------------------------------------------------------------------------##

